### DISCUSSION TITLE
Tags are empty when language different

### DISCUSSION THREAD

**Discussion by flexchar** [spatie/nova-tags-field#169] [Q&A] [answered]
I am using this and underlying tags package for managing tags in my application. Application is multilingual. I use _locale_ middleware for detecting /{locale_code} (where English is just plain root) and setting app locale for frontend. 

The site is being imported from WP. as such I have imported and programatically assigned values to respective places. This results in, say, posts in different locales w/tags respectively.

Briefly from **tags** table.
![image](https://user-images.githubusercontent.com/9198896/48899147-0fa66880-ee92-11e8-880d-589cf58da9e4.png)

Content appears normally when viewing on frontend. Note that _locale_ middleware uses `app()->setLocale()` to set locale.

However, when using Laravel Nova, which, obviously, doesn't use _locale_ middleware, **tags are not visible in other than English (default) locale**.

![image](https://user-images.githubusercontent.com/9198896/48899003-a9214a80-ee91-11e8-9887-6d8e17db8788.png)
![image](https://user-images.githubusercontent.com/9198896/48899279-85123900-ee92-11e8-84cf-a4e527364461.png)

I imagine this is somewhat designed (by original package) behaviour. But it doesn't make much sense w/Nova, where localization is not necessary. Any ideas? Do you think it is something you could improve? Or perhaps any suggestions for local workaround?

Thanks for the good work!

**Comment by flexchar** [comment, marked as answer]
I've got an idea!

One is to set Laravel locale to entry's locale when retrieving. It can be achieved by registering `retrieved` event listener in `boot` method in `app\Providers\NovaServiceProvider.php`.

![image](https://user-images.githubusercontent.com/9198896/48901329-7464c180-ee98-11e8-9ef7-fea706d9725e.png)

Nova::serving limits scope to Nova request only, thus avoiding registering at other times.

**Comment by freekmurze** [comment]
@flexchar Could you PR this?

**Comment by flexchar** [comment]
@freekmurze to be honest, I am not sure how should the issue be solved...

My approach is application specific as I have locale field/column on my model's table.

While I could imagine this showing all tag translations, it would create more mess than help.

**Comment by mansoorkhan96** [comment]
Having same issue.
